src/core.rs

```diff
@@ -235,15 +258,25 @@ impl std::ops::Deref for Word<'_> {
 impl<'a> Word<'a> {
     /// Construct a `Word` from a string.
     ///
-    /// A trailing stretch of `' '` is automatically taken to be the
+    /// All trailing whitespace is automatically removed to be the
     /// whitespace part of the word.
     pub fn from(word: &str) -> Word<'_> {
-        let trimmed = word.trim_end_matches(' ');
+        Self::with_tab_width(word, 0)
+    }
+
+    /// Construct a `Word` from a string, with a customizable tab width
+    /// to calculate the total width with.
+    ///
+    /// All trailing whitespace is automatically removed to be the
+    /// whitespace part of the word.
+    pub fn with_tab_width(word: &str, tab_width: u8) -> Word<'_> {
+        let trimmed = word.trim();
         Word {
             word: trimmed,
-            width: display_width(trimmed),
+            width: display_width_with_tab(trimmed, tab_width),
```

Comment:
There shouldn't be any `'\t'` characters in `trimmed`, right? So we should be able to measure with the normal `display_width`.

Reply:
That's very fair, thanks for pointing that out - will fix.